tune_sklearn/utils.py

```diff
@@ -30,6 +40,40 @@ def check_warm_start_ensemble(estimator):
             and hasattr(estimator, "n_estimators") and is_ensemble_subclass)
 
 
+def check_error_warm_start(estimator, estimator_config):
+    early_stop_type = get_early_stop_type(estimator)
+    if (early_stop_type == EarlyStopping.WARM_START_ITER
+            and "max_iter" in estimator_config):
+        raise ValueError("tune-sklearn uses `max_iter` to warm "
+                         "start, so this parameter can't be "
+                         "set when warm start early stopping. ")
+    if (early_stop_type == EarlyStopping.WARM_START_ENSEMBLE
+            and "n_estimators" in estimator_config):
```

Comment:
Also check for `max_iter` and `n_estimators` in pipeline config. Something like:
```python
if check_is_pipeline(estimator):
    if (early_stop_type == EarlyStopping.WARM_START_ENSEMBLE
            and f"{estimator.steps[-1][0]}__n_estimators" in estimator_config):
```

Reply:
Ah good catch

Reply:
Fixed in #106